pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -148,6 +148,11 @@
 
     private final ConnectionHandler connectionHandler;
 
+    // A batch timer task is only ever scheduled when a message is added to a message batch without also triggering a
```

Comment:
this is no more a "timer task". 
[javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ScheduledExecutorService.html#schedule(java.lang.Runnable,long,java.util.concurrent.TimeUnit))
schedule method returns "a ScheduledFuture representing pending completion of the task and whose get() method will return null upon completion"
scheduleAtFixedDelay returned "a ScheduledFuture representing pending completion of the series of repeated tasks. The future's [get()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/Future.html#get()) method will never return normally, and will throw an exception upon task cancellation or abnormal termination of a task execution."

For this reason, it would be better to rename `batchTimerTask` field accordingly. It could be simply `batchTask`.

Reply:
Good point. I think I'll rename it to `batchFlushTask`.